astropy/io/fits/column.py

```diff
@@ -1389,19 +1390,21 @@ def __getattr__(self, name):
 
         Implements for example self.units, self.formats, etc.
         """
-
         cname = name[:-1]
         if cname in KEYWORD_ATTRIBUTES and name[-1] == 's':
             attr = []
-            for col in self:
+            for col in self.columns:
                 val = getattr(col, cname)
-                if val is not None:
-                    attr.append(val)
-                else:
-                    attr.append('')
+                attr.append(val if val is not None else '')
             return attr
         raise AttributeError(name)
 
+    def _update_attrs(self, name=None):
+        """Get or update local lists of column attributes"""
+        attrs = [name] if name is not None else KEYWORD_ATTRIBUTES
+        for name in attrs:
+            setattr(self, name + 's', self.__getattr__(name+'s'))
```

Comment:
Fixed, though there is no real obligation for spaces around operators ;-)

> If operators with different priorities are used, consider adding whitespace around the operators with the lowest priority(ies). Use your own judgment; however, never use more than one space, ...

http://pep8.org/#other-recommendations